dashboard/models.py

```diff
@@ -150,14 +151,45 @@ def get_supported_courses(self):
             logger.info("Courses did not exist", exc_info = True)
         return []
 
+    def earliest_start_datetime(self) -> Optional[datetime.datetime]:
+        """Get the earliest start date of courses in the QuerySet
 
-class CourseManager(models.Manager):
-    def get_queryset(self):
-        return CourseQuerySet(self.model, using=self._db)
+        :return: Earliest start date of courses in the QuerySet, or None if no course dates found
+        :rtype: datetime
+        """
+        sorted_courses = sorted(self.all(), key=lambda course: course.course_date_range.start)
 
-    def get_supported_courses(self):
-        return self.get_queryset().get_supported_courses()
+        earliest_start = None
+        if len(sorted_courses) > 0:
+            earliest_course = sorted_courses[0]
+            earliest_start = earliest_course.course_date_range.start
+            logger.info(f"Earliest start datetime for all courses: {earliest_start.isoformat()} found in course {earliest_course.canvas_id}")
+        else:
+            logger.info(f"No course listed. Returning None as the earliest_start_datetime_of_course. ")
+        return earliest_start
 
+    def update_all_last_cron_run(self):
+        """ Updates the last cron run to now for all courses in QuerySet
+        """
+        # Finally update the time on all courses updated
+        self.update(last_cron_run=datetime.datetime.now())
+
+    def last_cron_run(self) -> Optional[datetime.datetime]:
+        """ Returns the datetime of the last cron run of all courses
+            This checks for any courses where the last_cron_run value is null.
+
+        :return: datetime. Either the earliest or None
+        :rtype: None
+        """
+        new_courses = self.filter(last_cron_run__isnull=True)
```

Comment:
This variable can be renamed `new_courses_from_last_cron_run`. It is bit long but it make sense  to me

Reply:
Okay fixed